custom_components/openplantbook/uploader.py

```diff
@@ -33,6 +33,87 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+# Take HASS state and verify if it is sane and supported by OPB and convert if necessary
+def get_supported_state_value(state) -> tuple:
+    def validate_measurement(supported_unit, value_range):
+        nonlocal state_error
+
+        if unit_of_measurement != supported_unit:
+            _LOGGER.debug(
+                "Unit '%s' of '%s' measurement is not supported. Its value '%s' disregarded"
+                % (unit_of_measurement, current_measurement, supported_state)
+            )
+            state_error = current_measurement
+
+        elif supported_state < value_range[0] or supported_state > value_range[1]:
+            _LOGGER.debug(
+                "Value '%s' of %s is out of range %s - disregarded"
+                % (supported_state, current_measurement, value_range)
+            )
+            state_error = current_measurement
+
+        return
+
+    current_measurement = state.attributes.get("device_class")
+    unit_of_measurement = state.attributes.get("unit_of_measurement")
+    state_error = None
+
+    try:
+        supported_state = round(float(state.state))
+    except:
+        _LOGGER.debug(
+            "State is not a number - disregarded: state_value: '%s', state: %s"
+            % (state.state, state)
+        )
+        return None, current_measurement
+
+    # temperature
+    if current_measurement == "temperature":
+
+        # Convert Fahrenheit to Celsius
+        if unit_of_measurement == "°F":
+            supported_state = round((supported_state - 32) * 5 / 9)
+            _LOGGER.debug(
+                "Temperature converted from %s °F to %s °C"
+                % (state.state, supported_state)
+            )
+            unit_of_measurement = "°C"
+
+        # Convert Kelvin to Celsius
+        elif unit_of_measurement == "K":
```

Comment:
Should this be "K" or °K"?

Reply:
Thanks for reviewing the code, mate. It is just "K" as per:

https://github.com/home-assistant/core/blob/be5577c2f9486167696aa7920cdb9e342cda2bf7/homeassistant/const.py
`
# Temperature units
class UnitOfTemperature(StrEnum):
    """Temperature units."""

    CELSIUS = "°C"
    FAHRENHEIT = "°F"
    KELVIN = "K"
`